### PR TITLE
Add new compute-firewall-policy-with-rules resource

### PR DESCRIPTION
**Pull request by mihhalj**
Add a new resource to allow updating all rules in firewall policy at once.

Fixes 
https://github.com/hashicorp/terraform-provider-google/issues/19195

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_compute_firewall_policy_with_rules` (beta)
```

### CODE REVIEW

**Comment by mihhalj** [comment]
Could I ask for review?

**Comment by rileykarson** [comment]
@slevenick do you mind taking this review? I've been spread a bit thin recently. There's some internal context if you need it (but hopefully it's clear in the context of the PR alone)

File: mmv1/products/compute/FirewallPolicyWithRules.yaml [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: FirewallPolicyWithRules
+min_version: beta
+base_url: 'locations/global/firewallPolicies?parentId={{parent}}'
+self_link: 'locations/global/firewallPolicies/{{policy_id}}'
+create_url: 'locations/global/firewallPolicies?parentId={{parent}}'
+update_verb: :PATCH
+description: "The Compute FirewallPolicy with rules resource"
```
Comment (slevenick) [inline mmv1/products/compute/FirewallPolicyWithRules.yaml:21]: Can we improve this description? Something like:
FirewallPolicyWithRules declaratively manages all rules attached to a firewall policy

or something that describes why a user would want to use this over the regular FirewallPolicy

File: mmv1/products/compute/FirewallPolicyWithRules.yaml [hunk 2]
```diff
@@ -0,0 +1,450 @@
...
+    description: The unique identifier for the resource. This identifier is defined by the server.
+    output: true
+    api_name: id
+  - !ruby/object:Api::Type::String
+    name: description
+    description: An optional description of this resource.
+  - !ruby/object:Api::Type::Array
+    name: 'rule'
+    api_name: 'rules'
+    description: A list of firewall policy rules. It must include pre-defined rules.
```
Comment (slevenick) [inline mmv1/products/compute/FirewallPolicyWithRules.yaml:72]: Does it need to include pre-defined rules? Don't they get split later?

File: mmv1/products/compute/FirewallPolicyWithRules.yaml [hunk 3]
```diff
@@ -0,0 +1,450 @@
...
+        - !ruby/object:Api::Type::Boolean
+          name: 'disabled'
+          description: |
+            Denotes whether the firewall policy rule is disabled. When set to true,
+            the firewall policy rule is not enforced and traffic behaves as if it did
+            not exist. If this is unspecified, the firewall policy rule will be
+            enabled.
+  - !ruby/object:Api::Type::Array
+    name: 'predefinedRules'
+    description: A list of firewall policy rules. It must include pre-defined rules.
```
Comment (slevenick) [inline mmv1/products/compute/FirewallPolicyWithRules.yaml:245]: Description here doesn't make sense, maybe:
A list of pre-defined rules created by the server. These have priority xxxxx-xxxxx

**Review by slevenick — CHANGES_REQUESTED** [review]
Sorry, missed this for a bit! Running tests now.

File: mmv1/products/compute/FirewallPolicyWithRules.yaml (hunk 2 above)
Comment (mihhalj) [reply mmv1/products/compute/FirewallPolicyWithRules.yaml:72]: right, fixed.

File: mmv1/products/compute/FirewallPolicyWithRules.yaml [hunk 4]
```diff
@@ -0,0 +1,450 @@
+# Copyright 2023 Google Inc.
```
[the remark this answers is not in the document]
Comment (mihhalj) [reply mmv1/products/compute/FirewallPolicyWithRules.yaml]: done

File: mmv1/templates/terraform/constants/resource_compute_firewall_policy_with_rules.go.erb [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      return intVal, nil
+    }
+  }
+
+  if intVal, ok := v.(int64); ok {
+    return intVal, nil
+  }
+
+  if floatVal, ok := v.(float64); ok {
+ 	  intVal := int64(floatVal)
```
[the remark this answers is not in the document]
Comment (mihhalj) [reply mmv1/templates/terraform/constants/resource_compute_firewall_policy_with_rules.go.erb]: done

File: mmv1/templates/terraform/constants/resource_compute_firewall_policy_with_rules.go.erb [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+	rules := make([]interface{}, 0)
+
+	for _, rule := range allRules {
+	  isPredefined, err :=  firewallPolicyWithRulesIsPredefinedRule(rule.(map[string]interface{}))
+	  if err != nil {
+	    return nil, nil, err
+	  }
+
+		if isPredefined {
+		 predefinedRules = append(predefinedRules, rule)
```
[the remark this answers is not in the document]
Comment (mihhalj) [reply mmv1/templates/terraform/constants/resource_compute_firewall_policy_with_rules.go.erb]: done

File: mmv1/products/compute/FirewallPolicyWithRules.yaml (hunk 1 above)
Comment (mihhalj) [reply mmv1/products/compute/FirewallPolicyWithRules.yaml]: done

File: mmv1/products/compute/FirewallPolicyWithRules.yaml (hunk 3 above)
Comment (mihhalj) [reply mmv1/products/compute/FirewallPolicyWithRules.yaml]: right, fixed

**Review by slevenick — APPROVED** [review]

**Pull request merged**